### PR TITLE
UIQM-50 Highlight rows when trash icon selected

### PR DESCRIPTION
**Pull request by uladislausamets**
## Purpose
https://issues.folio.org/browse/UIQM-50

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.css [hunk 1]
```diff
@@ -32,3 +32,7 @@
   width: 50px;
   margin-right: 10px;
 }
+
+.quickMarcFocusedRow {
+  background: rgba(114, 150, 226, 0.3);
```
Comment (aliaksei-chumakou) [inline src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.css:37]: Is there a specific requirement for this color? There is a similar named var `--color-fill-focus: rgba(37, 118, 195, 0.3);`

File: src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js [hunk 2]
```diff
@@ -45,6 +45,7 @@ const QuickMarcEditorRows = ({
 }) => {
   const [isRemoveModalOpened, toggleRemoveModal] = useModalToggle();
   const [removeIndex, setRemoveIndex] = useState();
+  const [focusedRowIndex, setfocusedRowIndex] = useState();
```
Comment (aliaksei-chumakou) [inline src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js:48]:
```suggestion
  const [focusedRowIndex, setFocusedRowIndex] = useState();
```

File: src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js [hunk 3]
```diff
@@ -128,7 +138,7 @@ const QuickMarcEditorRows = ({
                 }
               </div>
 
-              <div className={styles.quickMarcEditorRowTag}>
+              <div className={`${styles.quickMarcEditorRowTag} ${isFocusedRow ? styles.quickMarcFocusedRow : ''}`}>
```
Comment (NikitaSedyx) [inline src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js:141]: background should be applied for `[data-test-quick-marc-editor-row]`

File: src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js [hunk 4]
```diff
@@ -68,6 +69,14 @@ const QuickMarcEditorRows = ({
     });
   }, [moveRecord]);
 
+  const focusRow = useCallback(({ target }) => {
+    setfocusedRowIndex(+target.dataset.index);
+  }, [setfocusedRowIndex]);
```
Comment (NikitaSedyx) [inline src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js:74]: set function is not required in deps

File: src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js [hunk 5]
```diff
@@ -259,6 +269,8 @@ const QuickMarcEditorRows = ({
                           data-index={idx}
                           icon="trash"
                           onClick={showDeleteConfirmation}
+                          onFocus={focusRow}
```
Comment (NikitaSedyx) [inline src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js:272]: I don't think onFocus and onBlur will work as you should highlight row when modal is opened

**Comment by NikitaSedyx** [comment]
you should remove intlproviders and messages from the tests

File: src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js [hunk 6]
```diff
@@ -53,14 +54,22 @@ const QuickMarcEditorRows = ({
...
     setRemoveIndex(+target.dataset.index);
     toggleRemoveModal();
+    setFocusedRowIndex(+target.dataset.index);
   // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [setRemoveIndex]);
 
   const confirmDeletion = useCallback(() => {
     deleteRecord({ index: removeIndex });
     toggleRemoveModal();
+    setFocusedRowIndex(undefined);
```
Comment (aliaksei-chumakou) [inline src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js:64]: IMHO, undefined shouldn't be used as a value, there is a null to show that no value. undefined means more that it was never defined

**Review by aliaksei-chumakou — APPROVED** [review]

File: src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js [hunk 7]
```diff
@@ -53,14 +54,22 @@ const QuickMarcEditorRows = ({
...
   }, [deleteRecord, toggleRemoveModal, removeIndex]);
 
+  const onCloseDeleteConfirmation = useCallback(() => {
+    toggleRemoveModal();
+    setFocusedRowIndex(undefined);
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```
Comment (NikitaSedyx) [inline src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js:71]: seems toggle should be here

File: src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js [hunk 8]
```diff
@@ -53,14 +54,22 @@ const QuickMarcEditorRows = ({
...
   // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [setRemoveIndex]);
 
   const confirmDeletion = useCallback(() => {
     deleteRecord({ index: removeIndex });
     toggleRemoveModal();
+    setFocusedRowIndex(undefined);
   }, [deleteRecord, toggleRemoveModal, removeIndex]);
 
+  const onCloseDeleteConfirmation = useCallback(() => {
```
Comment (NikitaSedyx) [inline src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js:67]: closeDel...

File: src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.css [hunk 9]
```diff
@@ -32,3 +34,10 @@
   width: 50px;
   margin-right: 10px;
 }
+
+.quickMarcFocusedRow {
+  display: flex;
```
Comment (NikitaSedyx) [inline src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.css:39]: styles copy paste

**Comment by NikitaSedyx** [comment]
additionally please update test libs and react intl versions

**Review by AndreiShumski2 — APPROVED** [review]

File: package.json [hunk 10]
```diff
@@ -57,23 +57,23 @@
...
-    "jest": "^25.1.0",
+    "jest": "^26.4.2",
     "jest-css-modules": "^2.1.0",
-    "jest-junit": "^9.0.0",
+    "jest-junit": "^11.1.0",
     "miragejs": "^0.1.32",
     "react": "^16.5.1",
     "react-dom": "^16.5.1",
-    "react-intl": "^4.5.1",
+    "react-intl": "^5.8.1",
```
Comment (NikitaSedyx) [inline package.json:76]: peerDependencies not updated

File: src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js (hunk 6 above)
Comment (NikitaSedyx) [reply src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js:64]: any actions with it?

**Review by NikitaSedyx — APPROVED** [review]

**Pull request merged**